lang/py/src/avro/schema.py

```diff
@@ -722,14 +818,27 @@ def make_avsc_object(json_data, names=None):
   if hasattr(json_data, 'get') and callable(json_data.get):
     type = json_data.get('type')
     other_props = get_other_props(json_data, SCHEMA_RESERVED_PROPS)
+    logical_type = None
+    if 'logicalType' in json_data:
+      logical_type = json_data.get('logicalType')
+    if logical_type and logical_type != 'decimal':
+      raise SchemaParseException("Currently does not support %s logical type" % logical_type)
+    else:
```

Comment:
This `else` clause could be executed for the cases either it's not a logical type or it's a decimal logical type. This two `if` statements could be simplified to 
```
```
if 'logical_type' in json_data:
    logical_type = json_data.get('logicalType')
    if logical_type == 'decimal':
        precision = ...
        scale = ...
    else:
        raise ...
```